aws/ec2metadata/api.go

```diff
@@ -63,6 +64,42 @@ func (c *EC2Metadata) GetInstanceIdentityDocument() (EC2InstanceIdentityDocument
 	return doc, nil
 }
 
+// IAMInfo retrieves IAM info from the metadata API
+func (c *EC2Metadata) IAMInfo() (EC2IAMInfo, error) {
+	resp, err := c.GetMetadata("iam/info")
+	if err != nil {
+		return EC2IAMInfo{},
+			awserr.New("EC2RoleRequestError",
+				"failed to get EC2 IAM info", err)
+	}
+
+	info := EC2IAMInfo{}
+	if err := json.NewDecoder(strings.NewReader(resp)).Decode(&info); err != nil {
+		return EC2IAMInfo{},
+			awserr.New("SerializationError",
+				"failed to decode EC2 IAM info", err)
+	}
+
+	if info.Code != "Success" {
+		errMsg := fmt.Sprintf("Getting IAM Info was unsuccessful (%s)", info.Code)
+		return EC2IAMInfo{},
+			awserr.New("EC2RoleError",
+				errMsg, fmt.Errorf(errMsg))
```

Comment:
Can leave the third parameter of `awserr.New` nil. Since its an optional error param that available to set if there was an underlying error that `awserr.Error` is wrapping.